pep-9999.rst

```diff
@@ -0,0 +1,214 @@
+Author: Dustin Ingram <di@python.org>
+PEP-Delegate: Donald Stufft <donald@stufft.io>
+Status: Draft
+Type: Standards Track
+Topic: Packaging
+Content-Type: text/x-rst
+Created: 28-Dec-2022
+
+
+Abstract
+========
+
+:pep:`691` defined a JSON form for the "Simple Repository API". This allowed
+clients to more easily query the data that was previously only available in
+HTML, as defined in :pep:`503`.
+
+This proposal adds new fields to the JSON form, which allow it to be used in
+place of PyPI's `JSON API <https://warehouse.pypa.io/api-reference/json.html>`__
+to acquire data about known vulnerabilities for a given project.
+
+The new fields are all part of the data returned from the "project details" URL.
+
+The schema for the vulnerability data and related fields is based on the `Open
+Source Vulnerability Schema <https://ossf.github.io/osv-schema/>`__, provided
+by the `Open Source Vulnerability (OSV) <https://osv.dev/>`__ project of the
+`Open Source Software Security Foundation (OpenSSF) <https://openssf.org/>`__.
+
+
+Rationale
+=========
+
+With the introduction of the JSON form of the simple API in :pep:`691`, the
+simple API offers functionality that is almost as complete as the PyPI JSON API.
+This PEP adds data which were previously only available through the JSON API,
+in order to allow more clients which were previously Warehouse specific to
+support arbitrary standards-compliant indexes.
+
+
+Specification
+=============
+
+This specification defines version 1.2 of the simple repository API. For the
+HTML version of the API, there is no change from version 1.0. For the JSON
+version of the API, the following changes are made:
+
+- The ``api-version`` must specify version 1.2 or later.
+- A new ``vulnerabilities`` key is added at the top level.
+- One new "file information" key, ``vulnerability-ids`` is added to the
+  ``files`` data.
+
+The ``vulnerabilties`` and ``vulnerability-ids`` keys are mandatory.
+
+Vulnerabilities
+---------------
+
+An additional key, ``vulnerabilities`` MUST be present at the top level, in
+addition to the keys previous defined in :pep:`691` and :pep:`700`. This key
+MUST contain a mapping from a unique vulnerability identifier to a dictionary
+of fields that describe that vulnerability, for every vulnerability that
+affects the given project.
+
+An example of the structure for the ``vulnerabilities`` field:
+
+.. code-block:: json
+
+   "vulnerabilities": {
+       "PYSEC-2022-XXX": {
+           "aliases": [
+               "CVE-2022-XXXXX"
+           ],
+           "details": "A long description.",
+           "summary": "A shorter summary.",
+           "fixed_in": [
+               "1.2.3"
+           ],
+           "withdrawn": "2022-06-28T16:39:06Z"
+       }
+   }
+
+
+For this field, the key (vulnerability ID) MUST be unique, and MUST take the
+form of the `OSV ID field
+<https://ossf.github.io/osv-schema/#id-modified-fields>`__. The vulnerability
+ID prefix MUST be ``PYSEC-``.
```

Comment:
Just a heads up: I believe the `PYSEC-` prefix is not guaranteed in the current API responses: the OSV results that PyPI mirrors sometimes contain `OSV-` or `CVE-` IDs instead, if the vulnerability hasn't (yet) received a `PYSEC` ID.

Reply:
Ah, good catch. I'll fix this to say a `PYSEC-` prefix should be preferred if it's available.